spec.html

```diff
@@ -12308,50 +12241,35 @@ <h1>
         </h1>
         <dl class="header">
+          <dt>description</dt>
+          <dd>It returns a Boolean value which is *true* if and only if _Desc_ can be applied as the property of an object with specified _extensibility_ and current property _current_ while upholding <emu-xref href="#sec-invariants-of-the-essential-internal-methods">invariants</emu-xref>. When such application is possible and _O_ is not *undefined*, it is performed for the property named _P_ (which is created if necessary).</dd>
         </dl>
-        <emu-note>
-          <p>If *undefined* is passed as _O_, only validation is performed and no object updates are performed.</p>
-        </emu-note>
-        <p>It performs the following steps when called:</p>
         <emu-alg>
-          1. Assert: If _O_ is not *undefined*, then IsPropertyKey(_P_) is *true*.
+          1. Assert: ! IsPropertyKey(_P_) is *true*.
           1. If _current_ is *undefined*, then
             1. If _extensible_ is *false*, return *false*.
-            1. Assert: _extensible_ is *true*.
-            1. If IsGenericDescriptor(_Desc_) is *true* or IsDataDescriptor(_Desc_) is *true*, then
-              1. If _O_ is not *undefined*, create an own data property named _P_ of object _O_ whose [[Value]], [[Writable]], [[Enumerable]], and [[Configurable]] attribute values are described by _Desc_. If the value of an attribute field of _Desc_ is absent, the attribute of the newly created property is set to its <emu-xref href="#table-default-attribute-values">default value</emu-xref>.
+            1. If _O_ is *undefined*, return *true*.
+            1. If ! IsAccessorDescriptor(_Desc_) is *true*, then
+              1. Create an own accessor property named _P_ of object _O_ whose [[Get]], [[Set]], [[Enumerable]], and [[Configurable]] attribute values are described by _Desc_. If the value of an attribute field of _Desc_ is absent, the attribute of the newly created property is set to its <emu-xref href="#table-object-property-attributes">default value</emu-xref>.
             1. Else,
-              1. Assert: ! IsAccessorDescriptor(_Desc_) is *true*.
-              1. If _O_ is not *undefined*, create an own accessor property named _P_ of object _O_ whose [[Get]], [[Set]], [[Enumerable]], and [[Configurable]] attribute values are described by _Desc_. If the value of an attribute field of _Desc_ is absent, the attribute of the newly created property is set to its <emu-xref href="#table-default-attribute-values">default value</emu-xref>.
+              1. Create an own data property named _P_ of object _O_ whose [[Value]], [[Writable]], [[Enumerable]], and [[Configurable]] attribute values are described by _Desc_. If the value of an attribute field of _Desc_ is absent, the attribute of the newly created property is set to its <emu-xref href="#table-object-property-attributes">default value</emu-xref>.
             1. Return *true*.
           1. If every field in _Desc_ is absent, return *true*.
```

Comment:
```suggestion
          1. Assert: _current_ is a completed Property Descriptor.
          1. If every field in _Desc_ is absent, return *true*.
```
(altho we'll have to define what a completed property descriptor is, which is what CompletePropertyDescriptor returns)

Reply:
Is that the same as a "[fully populated Property Descriptor](https://tc39.es/ecma262/multipage/ecmascript-data-types-and-values.html#sec-property-descriptor-specification-type)"? 

> A fully populated Property Descriptor is one that is either an accessor Property Descriptor or a data Property Descriptor and that has all of the fields that correspond to the property attributes defined in either [Table 3](https://tc39.es/ecma262/multipage/ecmascript-data-types-and-values.html#table-data-property-attributes) or [Table 4](https://tc39.es/ecma262/multipage/ecmascript-data-types-and-values.html#table-accessor-property-attributes).

Reply:
aha, that's the phrase i was looking for :-) i'll update my suggestion.

Reply:
Done.